### PR TITLE
ROX-21647:  compliance profile results and stats

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

The profile results and counts.  Basically this provides the info for the following screen.  Example of the api call and results are in the test section.

Screenshot 2024-03-22 at 11 25 34 AM

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Ran this locally with multiple openshift clusters with CO installed.  This is what the API query is and the results of such query.

```
curl --insecure "https://central-stackrox.apps.ds-03-22-screeching-death-mi.ocp.infra.rox.systems:443/v2/compliance/scan/profile/results/ocp4-e8" -u admin:XXXX | json_pp
```

[e8_results.json](https://github.com/stackrox/stackrox/files/14724586/e8_results.json)

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/complianceoperator/v2/checkresults/datastore/stats_views.go [hunk 1]
```diff
@@ -43,3 +43,18 @@ type ResourceResultCountByProfile struct {
 	InconsistentCount  int    `db:"inconsistent_count"`
 	ProfileName        string `db:"compliance_profile_name"`
 }
+
+// ResourceResultsByProfile represents shape of the stats query for compliance operator results
+type ResourceResultsByProfile struct {
```
Comment (SimonBaeumer) [inline central/complianceoperator/v2/checkresults/datastore/stats_views.go:48]: As a user I might be interested in knowing when this result was created, so I know when the last scan was.
If UI wants to display it is on them. WDYTH?

File: central/complianceoperator/v2/checkresults/service/service_impl_test.go [hunk 2]
```diff
@@ -718,6 +718,87 @@ func (s *ComplianceResultsServiceTestSuite) TestGetComplianceProfileStats() {
...
+
+			results, err := s.service.GetComplianceProfileResults(s.ctx, tc.query)
+			if tc.expectedErr == nil {
+				s.Require().NoError(err)
+			} else {
+				s.Require().Error(tc.expectedErr, err)
+			}
+
+			if tc.expectedResp != nil {
+				s.Require().Equal(tc.expectedResp, results)
```
Comment (SimonBaeumer) [inline central/complianceoperator/v2/checkresults/service/service_impl_test.go:796]: Is it possible to return an error and a response?
If we can add `s.Require().NoError(err)`

**Review by SimonBaeumer — APPROVED** [review]

**Review by bradr5 — APPROVED** [review]

File: central/complianceoperator/v2/checkresults/datastore/stats_views.go (hunk 1 above)
Comment (dashrews78) [reply central/complianceoperator/v2/checkresults/datastore/stats_views.go:48]: Looks like that is in the mocks, but that would not be this endpoint.

Screenshot 2024-03-26 at 11 13 17 AM

Comment (dashrews78) [reply central/complianceoperator/v2/checkresults/datastore/stats_views.go:48]: https://issues.redhat.com/browse/ROX-23356

**Pull request merged**